### PR TITLE
releng: Add documentation Dockerfile

### PR DESCRIPTION
**Pull request by yaqwsx**

### CODE REVIEW

File: Dockerfile.doc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    git clone https://github.com/doxygen/doxygen.git /tmp/doxygen; \
+        cd /tmp/doxygen; \
+        git checkout 9b262a41ec2b1bcc2b64f7364730d164eeb3f2f5; \
+        mkdir build; \
+        cd build; \
+        cmake .. -DCMAKE_BUILD_TYPE=Release; \
+        make -j8; \
+        make install; \
+        rm -rf /tmp/doxygen \
+    apt-get remove gcc g++ build-essential
```
Comment (Ondraceq) [inline Dockerfile.doc:24]: Is there a reason to run `apt-get remove`?
Specifically confusing is the `remove build-essential`.

File: Dockerfile.doc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# We use bash instead of /bin/sh as ESP-IDF doesn't support /bin/sh
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        apt-utils wget software-properties-common gnupg
+
+RUN env DEBIAN_FRONTEND=noninteractive \
+        apt-get update; \
```
Comment (Ondraceq) [inline Dockerfile.doc:11]: Is `apt-get update` necessary? It is run in the previous step

File: Dockerfile.doc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:22.04
```
Comment (Ondraceq) [inline Dockerfile.doc:1]: The default for the original `Dockerfile` is `debian:bullseye`. Is there a reason to use `ubuntu:22.04` instead?

File: Dockerfile.doc (hunk 1 above)
Comment (yaqwsx) [reply Dockerfile.doc:24]: To shrink the image. You no longer build C++ in the container, so you don't need them. A better solution would be to use build-stages by Dockerfile, but they weren't available on the target systems (now they are, but this is fairly simple Dockerfile).

File: Dockerfile.doc (hunk 2 above)
Comment (yaqwsx) [reply Dockerfile.doc:11]: It is there to work fine with layer caching. It is not an expensive step. If you cache the previous operation and then build a month or two later, `apt-get install` will fail as the APT index would be out-of-date.

File: Dockerfile.doc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+# We use bash instead of /bin/sh as ESP-IDF doesn't support /bin/sh
```
[the remark this answers is not in the document]
Comment (yaqwsx) [reply Dockerfile.doc]: Fair point. I just used Ubuntu as it is my "go-to base image", but you are right, in the context of RoFI, we target Debian.

[the remark this answers is not in the document]
Comment (yaqwsx) [reply Dockerfile.doc]: You are right.

File: Dockerfile.doc (hunk 3 above)
Comment (yaqwsx) [reply Dockerfile.doc]: Fair point - we target Debian so we should use it also for documentation.

File: Dockerfile.doc (hunk 2 above)
Comment (Ondraceq) [reply Dockerfile.doc:8]: Then the question becomes why this is not in the original Dockerfile

**Review by Ondraceq — APPROVED** [review]
We should add `apt-get update` also to the normal `Dockerfile` on the next step (reasons in the above comments).

**Pull request merged**